File: backend/app/schemas/imports.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ImportJobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    download_task_id: int
    status: str
    mode: str
    total_files: int
    completed_files: int
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class ImportFileActionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    import_job_id: int
    download_file_id: int
    source_path: str
    target_path: str
    action_type: str
    status: str
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class ImportRequest(BaseModel):
    mode: str = Field(default="hardlink", pattern="^(hardlink|copy)$")


class ImportJobListResponse(BaseModel):
    message: str
    imports: list[ImportJobResponse]


class ImportJobMessageResponse(BaseModel):
    message: str
    import_job: ImportJobResponse


class ImportJobDetailResponse(BaseModel):
    message: str
    import_job: ImportJobResponse
    actions: list[ImportFileActionResponse]


class ImportSimpleMessageResponse(BaseModel):
    message: str

File: frontend/src/types/imports.ts
export type ImportJobStatus = 'pending' | 'completed' | 'failed' | 'rolled_back'
export type ImportMode = 'hardlink' | 'copy'
export type ImportActionStatus = 'pending' | 'completed' | 'failed' | 'rolled_back'
export type ImportActionType = 'hardlink' | 'copy'

export interface ImportJob {
  id: number
  download_task_id: number
  status: ImportJobStatus
  mode: ImportMode
  total_files: number
  completed_files: number
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface ImportFileAction {
  id: number
  import_job_id: number
  download_file_id: number
  source_path: string
  target_path: string
  action_type: ImportActionType
  status: ImportActionStatus
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface ImportJobListResponse {
  message: string
  imports: ImportJob[]
}

export interface ImportJobMessageResponse {
  message: string
  import_job: ImportJob
}

export interface ImportJobDetailResponse {
  message: string
  import_job: ImportJob
  actions: ImportFileAction[]
}

export interface ImportSimpleMessageResponse {
  message: string
}
